test/plugin.test.js

```diff
@@ -1,51 +1,33 @@
-var expect = require("expect");
-var installer = require("../src/installer");
-var Plugin = require("../src/plugin");
-var util = require("util");
-var webpack = require("webpack");
+var expect = require('expect');
+var installer = require('../src/installer');
+var Plugin = require('../src/plugin');
+var util = require('util');
+var webpack = require('webpack');
 
-describe("plugin", function() {
+describe('plugin', function() {
   beforeEach(function() {
-    this.check = expect.spyOn(installer, "check").andCall(function(dep) {
+    this.check = expect.spyOn(installer, 'check').andCall(function(dep) {
       return dep;
     });
 
-    this.checkBabel = expect.spyOn(installer, "checkBabel");
-
-    this.compiler = {
-      // Webpack >= 2 will reject config without an entry
-      options: {
-        entry: function() { return {} }
-      },
-      plugin: expect.createSpy().andCall(function(event, cb) {
-        if (event === "after-resolvers") {
-          cb(this.compiler);
-        }
-      }.bind(this)),
-      resolvers: {
-        loader: {
-          plugin: expect.createSpy(),
-          resolve: expect.createSpy(),
-        },
-        normal: {
-          plugin: expect.createSpy(),
-          resolve: expect.createSpy(),
-        },
-      },
-    };
```

Comment:
@evilebottnawi this was the problem. 
this test was passing the compiler object manually to the plugin.
now in webpack 4, the compiler instance is quite big and manually writing is not good.
So I used `webpack` to get the compiler, But the problem here is I cant put spies inside the its property cause its coming from webpack.

So I was thinking of doing non mocking/spying tests.
is it fine ?